### PR TITLE
RSA support for generated keys.

### PR DESCRIPTION
**Pull request by lukehinds**
Signed-off-by: Dan Lorenc <dlorenc@google.com>

### CODE REVIEW

File: pkg/ca/ca.go [hunk 1]
```diff
@@ -48,13 +48,18 @@ func Client() *privateca.CertificateAuthorityClient {
 	return c
 }
 
-func CheckSignature(alg string, pub crypto.PublicKey, proof []byte, email string) error {
+func CheckSignature(pub interface{}, proof []byte, email string) error {
```
Comment (bobcallaway) [inline pkg/ca/ca.go:51]: I'd put pub back to `crypto.PublicKey`... 100% equivalent but more readable.

File: pkg/ca/ca_test.go [hunk 2]
```diff
@@ -32,25 +31,49 @@ func failErr(t *testing.T, err error) {
 	}
 }
 
-func TestCheckSignature(t *testing.T) {
+func TestCheckSignatures(t *testing.T) {
```
Comment (bobcallaway) [inline pkg/ca/ca_test.go:34]: add ECDSA into the name here?

**Review by bobcallaway — COMMENTED** [review]
2 minor nits

File: pkg/ca/ca_test.go (hunk 2 above)
Comment (dlorenc) [reply pkg/ca/ca_test.go]: Done

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**